app/services/db/establish_connection/read_replica_connection.rb

```diff
@@ -0,0 +1,39 @@
+module Db
+  module EstablishConnection
+    class ReadReplicaConnection
+      def call
+        return yield if Rails.env.test?
+        ActiveRecord::Base.establish_connection(read_replica_connection_params)
+        yield
+      ensure
+        ActiveRecord::Base.establish_connection(primary_connection_params)
```

Comment:
Is this statement OK in a test environment?

There's this error in CI in the report specs (which makes me think it's not a fluke)
```
     Failure/Error: return PG::Connection.new( *args )
     
     PG::ConnectionBad:
       could not connect to server: No such file or directory
       	Is the server running locally and accepting
       	connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?
```

maybe it's worth doing an explicit `begin...ensure` to make sure we only run this bit when we change connections?

```ruby
def call
  return yield if Rails.env.test?
  begin
    ActiveRecord::Base.establish_connection(read_replica_connection_params)
    yield
  ensure
    ActiveRecord::Base.establish_connection(primary_connection_params)
  end
end
```

Reply:
Ah yep, my b

Reply:
Looks like this worked! 🎉